From c1afad2529b89d1ac455ba06e60cf71e4b3032d6 Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <xypron.glpk@gmx.de>
Date: Tue, 30 Jul 2019 22:57:39 +0200
Subject: [PATCH 1/1] clk: meson: remove duplicate logic

First thing we check in meson_clk_set_rate_by_id() is current_rate == rate.
There is not need to check it again.

Signed-off-by: Heinrich Schuchardt <xypron.glpk@gmx.de>
---
 drivers/clk/meson/gxbb.c | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/drivers/clk/meson/gxbb.c b/drivers/clk/meson/gxbb.c
index 2cb53fb92d..abb5337e78 100644
--- a/drivers/clk/meson/gxbb.c
+++ b/drivers/clk/meson/gxbb.c
@@ -823,10 +823,7 @@ static ulong meson_clk_set_rate_by_id(struct clk *clk, unsigned long id,
 	case CLKID_MPLL1:
 	case CLKID_MPLL2:
 	case CLKID_CLK81:
-		if (current_rate != rate)
-			return -EINVAL;
-
-		return 0;
+		return -EINVAL;
 	case CLKID_VPU:
 		return meson_clk_set_rate_by_id(clk,
 				meson_mux_get_parent(clk, CLKID_VPU), rate,
-- 
2.20.1

